### ISSUE TITLE
Create an abstract class 'JsonSerialiseableTest' to be used by all classes that should be json serialisable

### ISSUE THREAD

**Issue by p013570** [gchq/Gaffer#1098]
This abstract class should have an abstract method:

```java
@Test
public abstract void shouldJsonSerialiseAndDeserialise();
```

We have several existing abstract classes that contain this method - these should be updated to extend the JsonSerialiseableTest class.

It would be useful to have some sort of default implementation if possible.

**Comment by m607123** [comment]
# Migration
For any class that should JSON serialise/deserialise, their respective test class/superclass will now need to extend "JSONSerialisationTest", implementing the "getTestObject()" method to simply return an instance of the class under test.
Instances of empty/simple classes under test are now handled by the JSONSerialisationTest, and as such only custom/complex implementations are required in your test class.

**Comment by p013570** [comment]
Merged into develop-1.0.0.
